Automattic/wp-calypso
# Media modal: Adjust search box to fit within container

The box shadow overflows the container on which it is applied so this change makes the element smaller.

#### Changes proposed in this Pull Request

This component is used in many places, anywhere there is a `section header` with an expanding search. You can find this is /pages, /posts, and the media library.

Our solution is to remove the overflow on `dialog__content` which is a generic class use in many modals. We targeted this in CSS so that it should only apply to this editor media modal.

#### Testing instructions



* https://wordpress.com/post
* insert image
* open media library
* click on the search icon to expand the search box

Try different versions of the media modal to see if it breaks anything:

- different plans
- Jetpack vs WordPress.com

Before

<img width="969" alt="Screen Shot 2019-06-12 at 6 41 03 PM" src="https://user-images.githubusercontent.com/618551/59393552-b32d6c80-8d41-11e9-9281-3bd73c51ca3a.png">

After

<img width="944" alt="Screen Shot 2019-06-12 at 7 45 40 PM" src="https://user-images.githubusercontent.com/618551/59395685-d14b9a80-8d4a-11e9-86e1-e2d622cea869.png">



Fixes #30974 

cc @eeeeevon13
